mne/io/meas_info.py

```diff
@@ -215,16 +358,24 @@ def _check_consistency(self):
         if len(missing) > 0:
             raise RuntimeError('bad channel(s) %s marked do not exist in info'
                                % (missing,))
-        chs = [ch['ch_name'] for ch in self['chs']]
-        if len(self['ch_names']) != len(chs) or any(
-                ch_1 != ch_2 for ch_1, ch_2 in zip(self['ch_names'], chs)) or \
-                self['nchan'] != len(chs):
-            raise RuntimeError('info channel name inconsistency detected, '
-                               'please notify mne-python developers')
         # make sure we have the proper datatypes
         for key in ('sfreq', 'highpass', 'lowpass'):
             if self.get(key) is not None:
                 self[key] = float(self[key])
+        # make sure channel names are unique
+        seen = set()
+        duplicates = set()
+        for name in self['ch_names']:
+            if name in seen:
+                duplicates.add(name)
+            else:
+                seen.add(name)
+        if len(duplicates) > 0:
+            raise RuntimeError('Channel names are not unique, found '
+                               'duplicates for: %s' % duplicates)
+
+    def __hash__(self):
+        return object_hash(self._store)
```

Comment:
does the hash of an old instance of `info` equal the hash of the equivalent new version with this change?